src/irmin-pack/ext.ml

```diff
@@ -127,6 +127,8 @@ module Maker (Config : Conf.S) = struct
         let branch_t t = t.branch
 
         let batch t f =
+          let readonly = Conf.readonly t.config in
+          if readonly then failwith "Can't batch an RO pack store";
```

Comment:
At various points, some particular exception is thrown "RO_not_allowed" or similar. Perhaps we should standardise on that? In general, if a function is intended for RW only, why not add a dummy argument `batch ~rw_only:() t f` to mark that the interface should not be called in RO instances? Similarly `~ro_only` could mark those interfaces (`sync`?) that are for RO instances only.